ext/datadog_profiling_native_extension/ruby_helpers.c

```diff
@@ -19,35 +24,78 @@ void ruby_helpers_init(void) {
   _id2ref_id = rb_intern("_id2ref");
   inspect_id = rb_intern("inspect");
   to_s_id = rb_intern("to_s");
+  new_id = rb_intern("new");
+}
+// Raises a NativeError exception with seperate telemetry-safe and detailed messages.
+void _raise_native_error(VALUE native_exception_class, const char *detailed_message, VALUE static_message) {
+  #ifdef DD_DEBUG
+    if (native_exception_class != eNativeRuntimeError &&
+        native_exception_class != eNativeArgumentError &&
+        native_exception_class != eNativeTypeError) {
+        rb_raise(rb_eRuntimeError, "[ddtrace] BUG: _raise_native_error called with an exception that might not support two error messages. "
+            "Must be one of eNativeRuntimeError, eNativeArgumentError, or eNativeTypeError, was: %s", rb_class2name(native_exception_class));
+    }
+  #endif
```

Comment:
TBH I don't think this check needs to be protected by `DD_DEBUG` -- anyway something is already wrong, and we're about to throw an exception, so throwing a different exception about a second bug seems fine too -- we just need to fix one before we see the next one.